src/Microsoft.Data.SqlClient/tests/tools/TDS/TDS.EndPoint/TDSServerParser.cs

```diff
@@ -69,15 +69,17 @@ public void Run()
                 {
                     case TDSMessageType.PreLogin:
                         {
-                            if (Session.Encryption == TDSEncryptionType.None)
+                            if ((Session.Encryption == TDSEncryptionType.None || Session.Encryption == TDSEncryptionType.Off) &&
+                                (MessageBeingReceived[0] as TDSPreLoginToken).Encryption != TDSPreLoginTokenEncryptionType.On)
+                            //if (Session.Encryption == TDSEncryptionType.None)
                             {
                                 (MessageBeingReceived[0] as TDSPreLoginToken).Encryption = TDSPreLoginTokenEncryptionType.None;
                             }
 
                             // Call into the subscriber to process the packet
                             responseMessages = Server.OnPreLoginRequest(Session, MessageBeingReceived);
-
-                            if (Session.Encryption == TDSEncryptionType.None)
+          
+                            if (Session.Encryption == TDSEncryptionType.None || Session.Encryption == TDSEncryptionType.Off)
```

Comment:
Applied changes so that when TDSEncryptionType.None is used TDSPreLoginTokenEncryptionType.None is set and when TDSEncryptionType.Off is used TDSPreLoginTokenEncryptionType.None is set.